### PR TITLE
SSS compression: dense -> generators

### PR DESCRIPTION
**Pull request by hsignarg**
New function to create SSS generators from a dense matrix.
Better test parameters to fix issue #366 .

### CODE REVIEW

File: fflas-ffpack/ffpack/ffpack.h [hunk 1]
```diff
@@ -2032,6 +2032,52 @@ namespace FFPACK { /* SSS */
                            typename Field::ConstElement_ptr W, size_t ldw,
                            typename Field::ConstElement_ptr D, size_t ldd,
                            typename Field::Element_ptr A, size_t lda);
+            /**
+         * @brief Computes a quasi-separable matrix A from its SSS generators
```
Comment (ClementPernet) [inline fflas-ffpack/ffpack/ffpack.h:2036]: the converse: computes the SSS generators of a dense quasi-separable matrix

File: fflas-ffpack/ffpack/ffpack_sss.inl [hunk 2]
```diff
@@ -259,105 +259,296 @@ namespace FFPACK{
...
+                FFPACK::applyP (Fi, FFLAS::FflasRight, FFLAS::FflasNoTrans, s,
+                                0, N - s - 1, H + s, N, q); // Apply permutation q to H
+                FFLAS::fassign (Fi, s, fs, H + s, N, V, ldv);
+
+                // Temporary matrix for storing lower triangular of pluq
+                typename  Field::Element_ptr Temp = FFLAS::fflas_new (Fi, 2 * s, s);
+        
+                for (size_t brow = 0; brow < k - 2; brow++)
+                    {
+                        //              std::cout << "brow = "<<brow << std::endl;
```
Comment (ClementPernet) [inline fflas-ffpack/ffpack/ffpack_sss.inl:451]: remove these debugging prints

File: fflas-ffpack/ffpack/ffpack_sss.inl [hunk 3]
```diff
@@ -259,105 +259,296 @@ namespace FFPACK{
...
+                                              H + s * N * (brow + 1) + s * (brow + 2), N); // Remove L
+                        FFPACK::applyP (Fi, FFLAS::FflasRight, FFLAS::FflasNoTrans, s,
+                                        0, N - s* (brow + 2) - 1, H + s * N * (brow + 1) + s * (brow + 2), N,
+                                        q); // Apply permutation q to H
+                        FFLAS::fassign (Fi, s, s, H + s * N * (brow + 1) + s * (brow + 2), N,
+                                        V + ldv * s * (brow + 1), ldv); /* Sould not cause any trouble even if
+                                                                           last block*/
+                        // if (brow != k - 3) // Sould not cause segfault now
+                        // // shifts \hat H down
+                        // FFLAS::fassign (Fi, s, N - s * (brow + 3),
```
Comment (ClementPernet) [inline fflas-ffpack/ffpack/ffpack_sss.inl:488]: if this fassign is no longer need, just remove it

File: fflas-ffpack/ffpack/ffpack.h (hunk 1 above)
Comment (hsignarg) [reply fflas-ffpack/ffpack/ffpack.h]: Fixed in [9dc4531]

File: fflas-ffpack/ffpack/ffpack_sss.inl (hunk 2 above)
Comment (hsignarg) [reply fflas-ffpack/ffpack/ffpack_sss.inl]: Fixed in [9dc4531]

File: fflas-ffpack/ffpack/ffpack_sss.inl (hunk 3 above)
Comment (hsignarg) [reply fflas-ffpack/ffpack/ffpack_sss.inl]: Fixed in [9dc4531]

**Comment by ClementPernet** [comment]
LGTM

**Pull request merged**